### PR TITLE
docs(agw): Adding Ubuntu AGW migration guide

### PR DESCRIPTION
**Pull request by ardzoht**
Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

## Summary

- Adding migration guide for Debian to Ubuntu AGWs on docs

## Test Plan

- `./create_docusaurus_website.sh` to validate changes are applied

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: docs/readmes/lte/migration_guide.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- `Root CA`
+> */var/opt/magma/tmp/certs/rootCA.pem*
+
+- `Hardware ID`
+> */etc/snowflake*
+
+- `Challenge Key`
+> */var/opt/magma/certs/gw_challenge.key*
+
+After copying these, gather the output of `show_gateway_info.py` as reference and follow the [Ubuntu Install Guide](https://docs.magmacore.org/docs/lte/deploy_install_ubuntu)
```
Comment (pshelar) [inline docs/readmes/lte/migration_guide.md]: the deployment link is changed on master, can you change deploy_install_ubuntu  to deploy_install.

File: docs/docusaurus/sidebars.json [hunk 2]
```diff
@@ -140,6 +140,7 @@
         "type": "subcategory",
         "label": "Access Gateway",
         "ids": [
+          "lte/migration_guide",
```
Comment (hcgatewood) [inline docs/docusaurus/sidebars.json]: Call this "upgrade_debian_to_ubuntu"

File: docs/readmes/lte/migration_guide.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+id: migration_guide
+title: AGW migration guide
+hide_title: true
+---
+
+# Access Gateway migration guide
+
+> NOTE: Only Ubuntu-based Access Gateways are supported starting from Magma release 1.6, Debian support is deprecated.
+
+For a smooth transition from a Debian to Ubuntu access gateway, it is advised to backup local configs including `RootCA`, `control_proxy.yml`, `Hardware ID` and `Challenge Key` files. It is also recommended to backup any static IPs or routes provisioned on the existing access gateway.
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]:
```suggestion
For a smooth transition from a Debian to Ubuntu access gateway, back up local configs including

- Gateway's hardware ID
- Gateway's long-term challenge key (`gw_challenge.key`)
- Gateway's control_proxy config (`control_proxy.yml`)
- Root CA certificate (`rootCA.pem`)
- Any static IPs or routes provisioned directly on the existing gateway
```

File: docs/readmes/lte/migration_guide.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+---
+id: migration_guide
+title: AGW migration guide
+hide_title: true
+---
+
+# Access Gateway migration guide
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]:
```suggestion
# Migrate AGWs from Debian to Ubuntu
```

File: docs/readmes/lte/migration_guide.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+---
+id: migration_guide
+title: AGW migration guide
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]:
```suggestion
title: Migrate AGWs to Ubuntu
```

File: docs/readmes/lte/migration_guide.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+hide_title: true
+---
+
+# Access Gateway migration guide
+
+> NOTE: Only Ubuntu-based Access Gateways are supported starting from Magma release 1.6, Debian support is deprecated.
+
+For a smooth transition from a Debian to Ubuntu access gateway, it is advised to backup local configs including `RootCA`, `control_proxy.yml`, `Hardware ID` and `Challenge Key` files. It is also recommended to backup any static IPs or routes provisioned on the existing access gateway.
+
+### Before migration
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]:
```suggestion
## Before migration
```

File: docs/readmes/lte/migration_guide.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+#### Copy the contents from the following file locations to a local backup storage:
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]:
```suggestion
Copy the contents from the following file locations to a local backup storage
```

File: docs/readmes/lte/migration_guide.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+- `Hardware ID`
+> */etc/snowflake*
+
+- `Challenge Key`
+> */var/opt/magma/certs/gw_challenge.key*
+
+After copying these, gather the output of `show_gateway_info.py` as reference and follow the [Ubuntu Install Guide](https://docs.magmacore.org/docs/lte/deploy_install_ubuntu)
+
+### After migration
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]:
```suggestion
## After migration
```

File: docs/readmes/lte/migration_guide.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+> NOTE: Only Ubuntu-based Access Gateways are supported starting from Magma release 1.6, Debian support is deprecated.
+
+For a smooth transition from a Debian to Ubuntu access gateway, it is advised to backup local configs including `RootCA`, `control_proxy.yml`, `Hardware ID` and `Challenge Key` files. It is also recommended to backup any static IPs or routes provisioned on the existing access gateway.
+
+### Before migration
+
+#### Copy the contents from the following file locations to a local backup storage:
+
+- `Control Proxy`
+> */var/opt/magma/configs/control_proxy.yml*
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]: For this and the ones below

Also, put them in the same order as the previous section

```suggestion
- Control proxy config: `/var/opt/magma/configs/control_proxy.yml`
```

File: docs/readmes/lte/migration_guide.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+- `Challenge Key`
+> */var/opt/magma/certs/gw_challenge.key*
+
+After copying these, gather the output of `show_gateway_info.py` as reference and follow the [Ubuntu Install Guide](https://docs.magmacore.org/docs/lte/deploy_install_ubuntu)
+
+### After migration
+
+- Once Magma is installed on the Access Gateway, copy the contents from local backup to configure the new access gateway
+- Restart `magmad` services to enable the new configs:
+> *service magma@magmad restart*
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]:
```suggestion
- Restart `magmad` services to enable the new configs: `service magma@magmad restart`
```

File: docs/readmes/lte/migration_guide.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+- The output of `show_gateway_info.py` should match the one captured before migration
+- Verify access gateway is checked-in to the existing orchestrator and configs are being streamed successfully:
+> *journalctl -fu magma@magmad*
```
Comment (hcgatewood) [inline docs/readmes/lte/migration_guide.md]:
```suggestion
- Verify access gateway is checked-in to the existing Orchestrator and configs are being streamed successfully: `journalctl -fu magma@magmad`
```

**Review by hcgatewood — APPROVED** [review]
Wording and uniformity

**Pull request merged**